test/controllers/howDoYouWantToPay.controller.test.ts

```diff
@@ -0,0 +1,167 @@
+
+    dissolution = generateDissolutionGetResponse()
+    dissolutionSession = generateDissolutionSession(COMPANY_NUMBER)
+    validator = mock(FormValidator)
+
+    when(sessionService.getAccessToken(anything())).thenReturn(TOKEN)
+    when(sessionService.getDissolutionSession(anything())).thenReturn(dissolutionSession)
+    when(paymentService.generatePaymentURL(TOKEN, anything(), anything())).thenResolve(REDIRECT_CARD_URL)
+    when(dissolutionService.getDissolution(TOKEN, dissolutionSession)).thenResolve(dissolution)
+  })
+
+  function initApp(): Application {
+    return createApp(container => {
+      container.rebind(DissolutionService).toConstantValue(instance(dissolutionService))
+      container.rebind(SessionService).toConstantValue(instance(sessionService))
+      container.rebind(PaymentService).toConstantValue(instance(paymentService))
+      container.rebind(FormValidator).toConstantValue(instance(validator))
+    })
+  }
+
+  describe('GET', () => {
+
+    it('should redirect to the search company page if the application has been paid for', async () => {
+      dissolution.application_status = ApplicationStatus.PAID
+
+      const app = initApp()
+
+      await request(app)
+        .get(HOW_DO_YOU_WANT_TO_PAY_URI)
+        .expect(StatusCodes.MOVED_TEMPORARILY)
+        .expect('Location', SEARCH_COMPANY_URI)
+
+      verify(sessionService.setDissolutionSession(anything(), anything())).never()
+    })
+
+    it('should display the how do you want to pay page if the application has not been paid for', async () => {
+      dissolution.application_status = ApplicationStatus.PENDING_PAYMENT
+
+      const app = initApp()
+
+      const res = await request(app)
+        .get(HOW_DO_YOU_WANT_TO_PAY_URI)
+        .expect(StatusCodes.OK)
+
+      const htmlAssertHelper: HtmlAssertHelper = new HtmlAssertHelper(res.text)
+
+      assert.isTrue(htmlAssertHelper.containsText('h1', 'How do you want to pay for the application?'))
+    })
+  })
+
+  describe('POST', () => {
+    it('should display how do you want to pay page with validation error when no payment type is selected', async() => {
+      const form = {
+        paymentType: ''
+      }
+
+      const error: ValidationErrors = generateValidationError('paymentType', 'Select how the payment will be made')
+
+      dissolution.application_status = ApplicationStatus.PENDING_PAYMENT
+
+      when(validator.validate(deepEqual(form), anything())).thenReturn(error)
+
+      const app = initApp()
+
+      const res = await request(app)
+        .post(HOW_DO_YOU_WANT_TO_PAY_URI)
+        .send(form)
+        .expect(StatusCodes.BAD_REQUEST)
+
+      const htmlAssertHelper: HtmlAssertHelper = new HtmlAssertHelper(res.text)
+
+      assert.isTrue(htmlAssertHelper.containsText('.govuk-error-summary__body', 'Select how the payment will be made'))
+    })
+
+    it('should redirect to the credit card payment page when the credit/debit card payment type is selected', async () => {
```

Comment:
you should also assert that a paymentStateUUID has been set using `assert.isDefined(updatedSession.paymentStateUUID)